### DISCUSSION TITLE
Lookup of QRZ.COM data for "/P", "M" etc. if only main record exists in QRZ.com

### DISCUSSION THREAD

**Discussion by m0urs** [magicbug/Cloudlog#1533] [Ideas / Feature Requests]
Often I do have stations working e.g. portable with a "/P" extension of their call sign. Some of these stations do not have a separate entry or alias for this extension defined so I do not get any information back from QRZ.com

I currently then do a lookup without the extension and then add the extension again.

I would like to have Cloudlog just to use the QRZ.com data from the main entry (without extension) if no data for the call with extension exists in QRZ.com.

Does this make sense? If yes, could we add that as a featrue request?

Thanks,
Michael (DG3NAB)

**Comment by phl0** [comment]
But wouldn't that be the other way round then for stations that do have an alias defined?

**Reply by m0urs** [reply]
Maybe I do not understand what you mean?

If a station has a record for e.g. "DG3NAB/P" in QRZ.com, then Cloudlog should use the information for this record.
If a station has no record for e.g. "DG3NAB/P" in QRZ.com, then Cloudlog should use the information for "DG3NAB".

If you enter a callsign without extension (means without "/" in the callsign) e.g. "DG3NAB", then it should use only the record for "DG3NAB".

If there is only a record for "DG3NAB/P" in QRZ.com (do not think that this is the case in reality) but not for "DG3NAB" then it should not do anything.

Is there a case I did miss?

**Reply by phl0** [reply]
Don't think so. But currently the link is just based on what you type into the callsign field. I think at that time we do not have info, whether this really exists or the callsign just has a base entry (i.e. without prefix).

**Reply by m0urs** [reply]
I mean at the time when you enter the tab key so that the information is read from QRZ.com and written into the fields (if a record exists). As I said I currently do this manually:

Enter e.g. "DG3NAB/P" and then "TAB". 
I see that no fields are auto filled (like name, QTH, locator etc.)
I go back, delete "/P", hit "TAB" again.
This time all information is correctly read from QRZ.COM and prefilled.
I go back, add the "/P" again and the prefilled fields are still there.

So that is what I like to have automated.

**Reply by phl0** [reply]
I will have a look if the API gives a hint for automating this.

**Reply by m0urs** [reply]
Thank you. But is this really dependent of the API? Couldn't just Cloudlog make a second request to QRZ.com if there is no result returnd from QRZ.com and the call sign contains a "/"? And makes the second request just without the part after "/"? 

It would much more cool if we could also take care of call signs like "PA/DG3NAB/P" and then try to do requests like

"PA/DG3NAB/P"
"DG3NAB/P"
"DG3NAB"

until a match is found. :-)

**Reply by phl0** [reply]
Hhmm it seems that I cannot even confirm what you reported. This is what I get after entering "DG3NAB/P" and hitting TAB:

![Screenshot from 2022-07-16 13-32-56](https://user-images.githubusercontent.com/7112907/179353118-81d0c53d-ca1c-41b3-b305-0254490abc57.png)

**Reply by m0urs** [reply]
Yea, I DO HAVE an entry for "DG3NAB/P" :-)

**Reply by m0urs** [reply]
Try "DF5FF/P"

**Reply by phl0** [reply]
I am worried. You wrote earlier:

> Enter e.g. "DG3NAB/P" and then "TAB".
> I see that no fields are auto filled (like name, QTH, locator etc.)

Did I miss something?

**Reply by phl0** [reply]
> Try "DF5FF/P"

cnfm. That gives empty results.

**Reply by m0urs** [reply]
Yeah, sorry. Bad idea to use my own callsign as an example :-( I just did not want to use someone elses call sign here.

**Reply by m0urs** [reply]
You can also use "DG3NAB/M" as there is no alias defined for that.
